addon/components/component-proxy.js

```diff
@@ -87,6 +87,7 @@ export default Component.extend({
 
     let propsToUnset = oldPropNames.reduce((propsToUnset, propName) => {
       propsToUnset[propName] = undefined;
+      return propsToUnset;
```

Comment:
No idea how this didn't blow up on us before, showed up in one of the new tests I added for this PR 😅